[WHO] Implement Everybody Lives!

padfoothelix opened the pull request:
Implement [[Everybody Lives!]].

Part of #10653.

I created several common files to implement this ; I am not sure if this is the preferred way. 
In GainAbilityAllPlayersEffect, I copied some logic from GainAbilityControllerEffect, including use of custom duration, but I'm not sure if it is necessary - how is it different from Duration.WhileOnBattlefield ?

As always, thanks for your work.

padfoothelix commented:
Hmm I have no idea why the tests failed during the first build… and why it worked on the second.

JayDi85 commented:
Cause on first run master contained a failed verify test: 17e0c82a075e8752763b78f1648df4ca3acffd20

padfoothelix commented:
Yes, but what I don't get is that after I pulled from remote my own master branch now gives me a different test error :

[INFO] Standard out
Ability text checks started for DFT

Wrong card 3320: Brightglass Gearhulk
+ First strike
+ Trample
- When Brightglass Gearhulk enters, you may search your library for up to two artifact, creature, and/or enchantment cards with mana value 1 or less, reveal them, put them into your hand, then shuffle.
ref:
+ First strike
+ Trample
- When this creature enters, you may search your library for up to two artifact, creature, and/or enchantment cards with mana value 1 or less, reveal them, put them into your hand, then shuffle.


Wrong card 3355: Earthrumbler
+ Crew 3
+ Trample
+ Vigilance
- Exile an artifact or creature card from your graveyard: Earthrumbler becomes an artifact creature in addition to its other types until end of turn.
ref:
+ Crew 3
+ Trample
+ Vigilance
- Exile an artifact or creature card from your graveyard: This Vehicle becomes an artifact creature until end of turn.

........................

[INFO] Standard out
Error: unknown official set: DRC - Aetherdrift Commander (make sure it use correct set code or mark it as SetType.CUSTOM_SET)
Error: unknown official set: FIC - Final Fantasy Commander (make sure it use correct set code or mark it as SetType.CUSTOM_SET)
Error: unknown official set: FIN - Final Fantasy (make sure it use correct set code or mark it as SetType.CUSTOM_SET)
Error: unknown official set: PW25 - Wizards Play Network 2025 (make sure it use correct set code or mark it as SetType.CUSTOM_SET)
Error: wrong booster settings (set MUST HAVEN'T booster, but have) - 2025 - DFT - Aetherdrift
[INFO] 
[INFO] Standard error
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   VerifyCardDataTest.test_checkMissingSetData:1095 Found set errors: 5
[ERROR]   VerifyCardDataTest.test_verifyCards:321 found 2 errors in 81315 cards verify (see errors list above)
[INFO] 
[ERROR] Tests run: 23, Failures: 2, Errors: 0, Skipped: 5

Compilation works with -DskipTests.
Any idea ?

JayDi85 commented:
Open `Mage.Verify` folder and delete two zip-files with mtgjson data. It will help to redownload actual data on next run.

padfoothelix commented:
Thanks ! That did it.

PurpleCrowbar commented:
Not sure how much of this needs to be in common files honestly, [Courageous Resolve](https://github.com/magefree/mage/blob/0ccc5a706b0069d4e86cf6181a6153b10b07fd4e/Mage.Sets/src/mage/cards/c/CourageousResolve.java) is a similar card with the loss of life prevention and it handles it all in the class itself. Curious to hear what others think

xenohedron commented:
I agree, it would be cleaner to have a single custom effect within the card file, and not make three "common" classes used only here. (Unless there are actually several similar cards that could benefit from common logic, but I don't get the sense that's the case here.)

padfoothelix commented:
I have removed the common and moved the effect to the card file.

Note that while doing so I noticed that [[Courageous Resolve]] was bugged. I fixed it.
